mock/py/mockbuild/backend.py

```diff
@@ -536,6 +546,62 @@ def chain(self, args, options, buildroot):
                 log.info(pkg)
         return return_code
 
+    def rpkgPreprocess(self, host_chrootspec, host_chrootsources, timeout):
+        rpkg_conf_path = os.path.join(host_chrootsources, 'rpkg.conf')
+
+        if not os.path.isfile(rpkg_conf_path):
+            self.buildroot.root_log.info(
+                "rpkg.conf not found. Skipping rpkg preprocessing step.")
+            return
+
+        parser = configparser.ConfigParser(
+            interpolation=configparser.ExtendedInterpolation())
+
+        try:
+            parser.read(rpkg_conf_path)
+        except Exception as e:
```

Comment:
Does it really need to be wide general Exception? The same few lines below.

Reply:
Yes, I think I could be using `configparser.ParsingError` here and on the few lines below:
```
        try:
            preprocess_spec = parser.getboolean('rpkg', 'preprocess_spec')
        except:
            preprocess_spec = False
```
I can use more generic `configparser.Error`. I will prepare a patch. I will also fix the linting errors.

Reply:
Except for `configparser.Error`, I also needed to check for `ValueError` which is raised when `rpkg.preprocess_spec` is not a boolean. I also added a log message when we couldn't read the value from `rpkg.conf`. Please, see the new commits.